src/test/globalSetup.test.ts

```diff
@@ -106,6 +109,16 @@ export const mochaHooks = {
         globals.codelensRootRegistry.dispose()
         globalSandbox.restore()
 
+        // Don't run any validations for tests on telemetry
+        if (!(this.currentTest ?? this.test)?.file?.includes('shared/telemetry/')) {
+            try {
+                validateTelemetryMetrics()
```

Comment:
https://github.com/aws/aws-toolkit-vscode/blob/235a6a67d5ada3e6a66748490d49a575bc13c70c/src/shared/telemetry/telemetryService.ts#L149

There we can skip the check if `!isAutomation()`, see also https://github.com/aws/aws-toolkit-vscode/blob/235a6a67d5ada3e6a66748490d49a575bc13c70c/src/shared/telemetry/activation.ts#L58-L61